src/theme/src/hooks/useCheckboxAppearance.js

```diff
@@ -1,140 +1,35 @@
-const checkedDisabledState =
-  '&[disabled]:checked + div, &[type=checkbox][disabled]:indeterminate + div'
-
-const hiddenCheckboxStyle = {
-  border: '0',
-  clip: 'rect(1px, 1px, 1px, 1px)',
-  height: '1',
-  overflow: 'hidden',
-  padding: '0',
-  position: 'absolute',
-  whiteSpace: 'nowrap',
-  width: '1',
-  opacity: '0'
-}
-
-const baseStyle = {
-  WebkitFontSmoothing: 'antialiased',
-  textDecoration: 'none',
-  WebkitAppearance: 'none',
-  MozAppearance: 'none',
-  border: 'none',
-  outline: 'none',
-  cursor: 'pointer'
-}
-
-const checkedStyle = {
-  '& > svg': {
-    display: 'block'
-  }
-}
-/**
- * There is only a single appearance in the default theme.
- * @param {String} appearance.
- * @return {Object} the appearance of the checkbox.
- */
-const getCheckboxStyles = theme => {
-  const {
-    tokens: { primary, colors },
-    checkbox
-  } = theme
-
-  const {
-    base: baseStyles = {},
-    disabled: disabledStyles = {},
-    hover: hoverStyles = {},
-    focus: focusStyles = {},
-    active: activeStyles = {},
-    checked: checkedStyles = {},
-    checkedHover: checkedHoverStyles = {},
-    checkedDisabled: checkedDisabledStyles = {},
-    checkedActive: checkedActiveStyles = {}
-  } = checkbox || {}
-
-  return {
-    ...hiddenCheckboxStyle,
-    '& + div > svg': { display: 'none' },
-    [defaultState]: {
-      ...baseStyle,
-      color: 'white',
-      backgroundColor: 'white',
-      boxShadow: `inset 0 0 0 1px ${colors.gray400}`,
-      ...baseStyles
-    },
-    [disabledState]: {
-      cursor: 'not-allowed',
-      backgroundColor: colors.gray100,
-      boxShadow: `inset 0 0 0 1px ${colors.gray100}`,
-      ...disabledStyles
-    },
-    [hoverState]: {
-      boxShadow: `inset 0 0 0 1px ${colors.gray600}`,
-      ...hoverStyles
-    },
-    [focusState]: {
-      boxShadow: `0 0 0 2px ${colors.blue100}`,
-      ...focusStyles
-    },
-    [activeState]: {
-      backgroundColor: colors.gray100,
-      boxShadow: `inset 0 0 0 1px ${colors.gray500}`,
-      ...activeStyles
-    },
-    [checkedState]: {
-      ...checkedStyle,
-      color: 'white',
-      boxShadow: `inset 0 0 0 -1px ${primary.active}`,
-      backgroundColor: primary.base,
-      ...checkedStyles
-    },
-    [checkedHoverState]: {
-      ...checkedStyle,
-      color: 'white',
-      backgroundColor: primary.hover,
-      boxShadow: `inset 0 0 0 1px ${primary.hover}`,
-      ...checkedHoverStyles
-    },
-    [checkedDisabledState]: {
-      ...checkedStyle,
-      color: colors.gray600,
-      backgroundColor: colors.gray100,
-      ...checkedDisabledStyles
-    },
-    [checkedActiveState]: {
-      ...checkedStyle,
-      color: 'white',
-      boxShadow: `inset 0 0 0 -1px ${primary.active}`,
-      backgroundColor: primary.active,
-      ...checkedActiveStyles
-    }
-  }
+const pseudoSelectors = {
+  _display: '& + div',
+  _tickSvg: '& + div > svg',
```

Comment:
Hm, this presents a challenge for how we handle subcomponent styling. I don't know that aliasing a selector is the right move, but happy to move forward with this to unblock things for now. It feels non-standard.